ImperatorToCK3Tests/MapperTests/ReligionMapper/ReligionMappingTests.cpp

```diff
@@ -1,49 +1,145 @@
-#include "../../ImperatorToCK3/Source/Mappers/ReligionMapper/ReligionMapping.h"
+#include "Mappers/ReligionMapper/ReligionMapping.h"
+#include "Mappers/RegionMapper/CK3RegionMapper.h"
+#include "Mappers/RegionMapper/ImperatorRegionMapper.h"
 #include "gtest/gtest.h"
 #include <sstream>
 
 
-TEST(Mappers_ReligionMappingTests, ck3ReligionDefaultsToBlank)
+
+TEST(Mappers_ReligionMappingTests, regularMatchOnSimpleReligion)
 {
 	std::stringstream input;
-	input << "= {}";
+	input << "ck3 = flemish imp = dutch";
+	const mappers::ReligionMapping theMapping(input);
+
+	ASSERT_EQ("flemish", *theMapping.religionMatch("dutch", 56, 445));
+}
 
+TEST(Mappers_ReligionMappingTests, regularMatchOnSimpleReligionFailsForWrongCulture)
+{
+	std::stringstream input;
+	input << "ck3 = flemish imp = flemish";
 	const mappers::ReligionMapping theMapping(input);
 
-	ASSERT_TRUE(theMapping.getCK3Religion().empty());
+	ASSERT_FALSE(theMapping.religionMatch("dutch", 345, 435));
 }
 
 
-TEST(Mappers_ReligionMappingTests, ck3ReligionCanBeSet)
+TEST(Mappers_ReligionMappingTests, matchOnProvince)
 {
 	std::stringstream input;
-	input << "= { ck3 = ck3Religion }";
-
+	input << "ck3 = dutch imp = german ck3Province = 17";
 	const mappers::ReligionMapping theMapping(input);
 
-	ASSERT_EQ("ck3Religion", theMapping.getCK3Religion());
+	ASSERT_EQ("dutch", *theMapping.religionMatch("german", 17, 45));
 }
 
-
-TEST(Mappers_ReligionMappingTests, imperatorReligionsDefaultToEmpty)
+TEST(Mappers_ReligionMappingTests, matchOnProvinceFailsForWrongProvince)
 {
 	std::stringstream input;
-	input << "= {}";
+	input << "ck3 = dutch imp = german ck3Province = 17";
+	const mappers::ReligionMapping theMapping(input);
+
+	ASSERT_FALSE(theMapping.religionMatch("german", 19, 345));
+}
 
+TEST(Mappers_ReligionMappingTests, matchOnProvinceFailsForNoProvince)
+{
+	std::stringstream input;
+	input << "ck3 = dutch imp = german ck3Province = 17";
 	const mappers::ReligionMapping theMapping(input);
 
-	ASSERT_TRUE(theMapping.getImperatorReligions().empty());
+	ASSERT_FALSE(theMapping.religionMatch("german", 0, 345));
+}
+
+TEST(Mappers_ReligionMappingTests, matchOnRegion)
+{
+	auto theMapper = std::make_shared<mappers::CK3RegionMapper>();
+	CK3::LandedTitles landedTitles;
+	std::stringstream landedTitlesStream;
+	landedTitlesStream << "k_ugada = { d_wakaba = { c_athens = { b_athens = { province = 79 } b_newbarony = { province = 56 } } } } \n";
```

Comment:
I happen to find the Builder thing I do in my own tests more readable.

Regardless, remove unnecessary details.